xarray/core/dataarray.py

```diff
@@ -5558,7 +5558,6 @@ def map_blocks(
         ...     gb = da.groupby(groupby_type)
         ...     clim = gb.mean(dim="time")
         ...     return gb - clim
-        ...
         >>> time = xr.cftime_range("1990-01", "1992-01", freq="ME")
```

Comment:
FWIW I really dislike this change, and find it makes the doctests much harder to scan. But a) I think it's maybe just me?, b) it's not enough to stop moving to ruff.

Unless there's some way to disable this? Even if we disabled doctest formatting in favor of retaining `blackdoc` (which doesn't have the same performance requirements as formatting the 7000 line python files)

Reply:
> I think it's maybe just me?

You're not alone. The sad thing is, this change is not consistent: we have a bunch of these in `xarray.core.computation` as well, which have not been removed (not sure why, though).

I'd be cautious about selectively enabling `ruff format` without docstrings: `blackdoc` relies on `black` for the formatting, so whenever `ruff` is (inevitably) inconsistent, we'll get inconsistent formatting between doctests and the normal code.

Reply:
> we'll get inconsistent formatting between doctests and the normal code.

Is this bad though? The differences are so so small, and if we can disable ruff from the docstrings then they're not going to step on each other?

Reply:
probably not too much

Reply:
OK, @Armavica if you're up for moving back to `blackdoc` for the docstrings, and reverting the docstring changes, then I'm +1 on merging; I'll merge unless anyone has objections

Reply:
looks like `ruff format` does not format the docstring of `apply_ufunc`, at all. Not sure why that is, other functions within `xarray.core.computation` are formatted just fine.

If we're truly keeping `blackdoc` it might make sense to drop the pin for `black` (with all the reproducibility issues that would bring): there's no way to sync it with the `black` hook anymore, so we'd have to upgrade manually.

Reply:
Ok, if I understand correctly the reason(s) for not using `ruff` on the docstrings are that:
- we would rather it kept the empty `...` after a function definition
- it's actually inconsistent on this issue

Is that right? If so, would you like me to create a new issue or PR to record this, and eventually come back to it when ruff improves on this front?

Reply:
yes to the point, we'll probably have to ask the `ruff` team to add an option for that, as I don't see a lot of other projects that prefer the trailing `...` after blocks – but I didn't ask around, so who knows.

For the second point we'll probably have to figure out why `ruff` does not format the docstring of `apply_ufunc`, at all. `blackdoc` formats this correctly, so it can't be caused by a `SyntaxError`.

Reply:
And to set the stakes: having ruff format the docstrings rather than `blackdoc` doesn't help us much — the main advantage of ruff is its speed vs. black, so saving a huge file doesn't take a few seconds. `blackdoc` doesn't have that much code to format and doesn't run on every save, so it's fine if it's slower.

It reduces one config, but that's fine to hold onto for a while.